lib/Core/Memory.cpp

```diff
@@ -920,3 +1032,38 @@ std::string PersistentState::getLocationInfo(const ExecutionState &state) {
 
   return infoStr;
 }
+
+std::string PersistentState::getAllocInfo(ref<Expr> offset, 
+                                          const char *type) const {
+  std::string infoStr;
+  llvm::raw_string_ostream msg(infoStr);
+  msg << "Persistent Memory Info:\n";
+  msg << "\tName: " << getObject()->name << "\n";
+  msg << "\tAddress: " << llvm::format_hex(getObject()->address, 18) << "\n";
+  msg << "\t\tOffset: " << *offset << "\n";
+  msg << "\tSize: " << getObject()->size << "\n";
+  msg << "\tType of modification: " << type << "\n";
+  msg << "\tAlloc at: ";
+  std::string tmp;
+  getObject()->getAllocInfo(tmp);
+  msg << tmp << "\n";
+
+  return infoStr;
+}
```

Comment:
Seems like duplicated code?

Reply:
Only a little. You have found my other laziness.